libraries/chain/webassembly/runtimes/eos-vm.cpp

```diff
@@ -420,7 +420,7 @@ REGISTER_LEGACY_HOST_FUNCTION(preactivate_feature, privileged_check);
 REGISTER_HOST_FUNCTION(set_resource_limits, privileged_check);
 REGISTER_LEGACY_HOST_FUNCTION(get_resource_limits, privileged_check);
 REGISTER_HOST_FUNCTION(get_parameters_packed, privileged_check);
-REGISTER_HOST_FUNCTION(set_parameters_packed, privileged_check);
+REGISTER_HOST_FUNCTION(set_parameters_packed, privileged_check, action_check); // Not allowed in sync calls. Change of max_sync_call_depth will interfere with active sync calls.
```

Comment:
Does this prevent any useful sub-divide of the system contract?

Reply:
I hated to break the uniformity of privileged host function preconditions too. The worse is this restriction is due to implementation of sync call wasm allocator queue resizing.

With the restriction, implementation would be simpler. We can resize right at when `set_parameters_packed` is executed.

Without the restriction, we know `set_parameters_packed` cannot be called from read-only threads and in read-window. On the main thread, we will need to check if `max_sync_call_depth` has changed. If changed, we resize the wasm allocator queue.

@spoonincode, @arhag What do you think about this?

Reply:
Personally I lean toward allowing it. The complication with resizing can be avoided by simply never shrinking the allocator vector/queue during operation. (yeah this leaves some resources unused until restart but seems no big deal)

Reply:
I think we should allow it.

Reply:
Thanks. I will revert to allow it.

Reply:
https://github.com/AntelopeIO/spring/pull/1311/commits/ed1f95544b7c71aa19b41d2e111ebda5dca108f6